modules/gdscript/gdscript_compiler.cpp

```diff
@@ -1742,6 +1742,7 @@ Error GDScriptCompiler::_parse_function(GDScript *p_script, const GDScriptParser
 				}
 				GDScriptCodeGenerator::Address dst_addr = codegen.parameters[parameter->identifier->name];
 				codegen.generator->write_assign(dst_addr, src_addr);
+				codegen.generator->set_default_parameter();
```

Comment:
I don't like this interface. A `set_*()` method without arguments is odd, hard to tell what it does.

In this case, I would prefer to create a function called `write_assign_default_parameter(Address dst, Address src)` and then replace the `write_assign()` call. This function could do what `set_default_parameter()` does then just call `write_assign` internally.

This will make more sense for other generators added in the future.

P.S. sorry for the delay reviewing this.

Reply:
@vnen done!